### ISSUE TITLE
Connection Pool Not Persisting Fetcharray mode

### ISSUE THREAD

**Issue by ulmont** [markdirish/node-odbc#199]
I'm using the fetcharray mode so that I can write consistent code whether I'm using ODBC or Postgres on the back end (ODBC ends up with objects having UPPERCASE names, Postgres ends up with objects having lowercase names, so using the object forms doesn't work).

...my code correctly returns arrays of arrays when pool.query is used if I use a single connection:

```javascript
  pool = await odbc.connect({
    connectionString: process.env.DATABASE_URL_DB2_ODBC,
    fetchArray: true // make sure we return these as arrays rather than objects, since the casing is different between databases
  })
```

...but incorrectly returns an array of objects if I use the connection pool with the same options:
```javascript
  pool = await odbc.pool({
    connectionString: process.env.DATABASE_URL_DB2_ODBC,
    fetchArray: true // make sure we return these as arrays rather than objects, since the casing is different between databases
  })
```

**Comment by ulmont** [comment]
I think the issue is that odbc.connect will pass the entire connection object to the native code:

```javascript
function connect(connectionString, callback) {
  if (typeof callback !== 'function') {
    return new Promise((resolve, reject) => {
      nativeOdbc.connect(connectionString, (error, odbcConnection) => {
        if (error) {
          reject(error);
        } else {
          const connection = new Connection(odbcConnection);
          resolve(connection);
        }
      });
    });
  }
```
But Pool, when it gets to increasePoolSize, will only use the connectionString, connectTimeout, and loginTimeout properties of the original connection object:

```javascript
class Pool {
  constructor(connectionString) {
    this.isOpen = false;
    this.freeConnections = [];

    // connectionString is a...
    if (typeof connectionString === 'string') {
      this.connectionString = connectionString;
      this.initialSize = INITIAL_SIZE_DEFAULT;
      this.incrementSize = INCREMENT_SIZE_DEFAULT;
      this.maxSize = MAX_SIZE_DEFAULT;
      this.shrink = SHRINK_DEFAULT;
      this.connectionTimeout = CONNECTION_TIMEOUT_DEFAULT;
      this.loginTimeout = LOGIN_TIMEOUT_DEFAULT;
    } else if (typeof connectionString === 'object') {
      const configObject = connectionString;
      if (!Object.prototype.hasOwnProperty.call(configObject, 'connectionString')) {
        throw new TypeError('Pool configuration object must contain "connectionString" key');
      }
      this.connectionString = configObject.connectionString;
      this.initialSize = configObject.initialSize || INITIAL_SIZE_DEFAULT;
      this.incrementSize = configObject.incrementSize || INCREMENT_SIZE_DEFAULT;
      this.maxSize = configObject.maxSize || MAX_SIZE_DEFAULT;
      this.shrink = configObject.shrink || SHRINK_DEFAULT;
      this.connectionTimeout = configObject.connectionTimeout || CONNECTION_TIMEOUT_DEFAULT;
      this.loginTimeout = configObject.loginTimeout || LOGIN_TIMEOUT_DEFAULT;
    } else {
      throw TypeError('Pool constructor must passed a connection string or a configuration object');
    }
  }
...
async increasePoolSize(count) {
    const connectionConfig = {
      connectionString: this.connectionString,
      connectionTimeout: this.connectionTimeout,
      loginTimeout: this.loginTimeout,
    };
    return new Promise((resolve, reject) => {
      const connectArray = [];
      for (let i = 0; i < count; i += 1) {
        connectArray.push((callback) => {
          odbc.connect(connectionConfig, (error, connection) => {
            if (!error) {
              if (this.isOpen) {
                this.freeConnections.push(new Connection(connection));
              }
            }
            callback(error, connection);
          });
        });
      }
```

**Comment by ulmont** [comment]
Pull request is here: https://github.com/markdirish/node-odbc/pull/200

**Comment by markdirish** [comment]
Was merged in with #200, which is now published in version 2.4.0. Feel free to open if it still seems broken!
